dashboard/app/models/programming_class.rb

```diff
@@ -55,20 +56,28 @@ def self.seed_all
 
   def self.seed_record(file_path)
     properties = properties_from_file(file_path, File.read(file_path))
-    record = ProgrammingClass.find_or_initialize_by(key: properties[:key], programming_environment_id: properties[:programming_environment_id])
-    record.assign_attributes(properties)
-    record.save! if record.changed?
-    record.id
+    programming_class = ProgrammingClass.find_or_initialize_by(key: properties[:key], programming_environment_id: properties[:programming_environment_id])
+    programming_class.assign_attributes(properties.except(:methods))
+    programming_class.save! if programming_class.changed?
+    programming_class.programming_methods =
```

Comment:
Is this assignment used?

Reply:
I guess not because `programming_class_id` is being set explicitly below. I can try to take it out and double check it still creates the methods correctly

Reply:
I tested that this still created the methods correctly but that methods were still deleted correctly. This line deletes any methods that aren't in the list, which allows for removing methods.

I'll fix this in the next PR but leaving this note as a TIL!